spec/rubocop/cop/rspec/preferred_shared_example_methods_spec.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::PreferredSharedExampleMethods, :config do
+  shared_examples 'preferred method' do |config_key, methods|
+    preferred = methods.last
+
+    context "with #{config_key}" do
+      let(:cop_config) do
+        { 'EnforcedStyle' => 'string', config_key => config_val }
+      end
+
+      context "when #{config_key} is unset" do
+        let(:config_val) { nil }
+
+        methods.each do |method|
+          it "does not register an offense for `#{method}`" do
+            expect_no_offenses(<<~RUBY)
+              #{method} 'desc'
+            RUBY
+          end
+
+          it "does not register an offense for `RSpec.#{method}`" do
+            expect_no_offenses(<<~RUBY)
+              RSpec.#{method} 'desc'
+            RUBY
+          end
+        end
+
+        it 'does not register an offense for another method' do
```

Comment:
This doesn’t check that the cop is not offended by ‘shared_state’/‘shared_scenario’, does it?

Reply:
I added `opposing_methods` to list the methods from the other group (example methods for context and vice versa), and ensured that they do not register an offense.